Matrix.py

```diff
@@ -0,0 +1,77 @@
+class Matrix:
+    def __init__(self, columns):
```

Comment:
Add a docstring and type annotations

Reply:
```suggestion
    def __init__(self: Self, init_data: ...):
        """Init method."""
```
the `Self` type comes from `typing` lib and you must use it to annotate the self type
this `...` **must** be the type of init_data that the new Matrix objects gets

also write the docstring to all methods and classes _but keep it simple_

Reply:
It is done